test/libsolidity/SolidityExecutionFramework.cpp

```diff
@@ -85,3 +81,28 @@ bytes SolidityExecutionFramework::compileContract(
 		cout << "metadata: " << m_compiler.metadata(contractName) << endl;
 	return obj.bytecode;
 }
+
+bytes SolidityExecutionFramework::compileContract(
+	string const& _sourceCode,
+	string const& _contractName,
+	map<string, Address> const& _libraryAddresses
+)
+{
+	return multiSourceCompileContract(
+		{{"", _sourceCode}},
+		_contractName,
+		_libraryAddresses
+	);
+}
+
+string SolidityExecutionFramework::addPreamble(string& _sourceCode)
```

Comment:
```suggestion
string SolidityExecutionFramework::addPreamble(string const& _sourceCode)
```
and then you'll also need to remove the `move()` below (it was only appropriate for the `string& io_sourceCode` version).

You could also keep the `move` and make the parameter `string _sourceCode` but that would only be better if you were modifying it in place. In my opinion `const&` is the best option here.

Reply:
sounds good.